### PR TITLE
Replaced Tracer.use_span() with opentelemetry.trace.use_span()

### LINKED ISSUE #1630: Make sure pymemcache instrumentation follows semantic conventions

### PR DESCRIPTION
**Pull request by owais**
# Description

Replaces `Tracer.use_span()` with `opentelemetry.trace.use_span()`.

Fixes https://github.com/open-telemetry/opentelemetry-python/issues/1630

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Updated existing tests

# Does This PR Require a Core Repo Change?

- [x] Yes. - Link to PR: https://github.com/open-telemetry/opentelemetry-python/pull/1668
- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-aiopg/src/opentelemetry/instrumentation/aiopg/aiopg_integration.py [hunk 1]
```diff
@@ -117,13 +116,7 @@ async def traced_execution(
...
         ) as span:
             self._populate_span(span, cursor, *args)
-            try:
-                result = await query_method(*args, **kwargs)
-                return result
-            except Exception as ex:  # pylint: disable=broad-except
-                if span.is_recording():
-                    span.set_status(Status(StatusCode.ERROR, str(ex)))
-                raise ex
+            return await query_method(*args, **kwargs)
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-aiopg/src/opentelemetry/instrumentation/aiopg/aiopg_integration.py:119]: The context manager above (`start_as_current_span`) handles exceptions and records them as status already so this was completely unnecessary/redundant. Some similar changes below.

File: instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py [hunk 2]
```diff
@@ -137,8 +137,8 @@ def _trace_prerun(self, *args, **kwargs):
             operation_name, context=tracectx, kind=trace.SpanKind.CONSUMER
         )
 
-        activation = self._tracer.use_span(span, end_on_exit=True)
-        activation.__enter__()
+        activation = trace.use_span(span, end_on_exit=True)
+        activation.__enter__()  # pylint: disable=E1101
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py:141]: pylint has false positives when using a generator as context manager so ignoring rule.

File: instrumentation/opentelemetry-instrumentation-aiopg/tests/test_aiopg_integration.py [hunk 3]
```diff
@@ -322,7 +320,7 @@ def test_span_failed(self):
         self.assertIs(
             span.status.status_code, trace_api.status.StatusCode.ERROR
         )
-        self.assertEqual(span.status.description, "Test Exception")
+        self.assertEqual(span.status.description, "Exception: Test Exception")
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-aiopg/tests/test_aiopg_integration.py:323]: Instrumentation was duplicating start_as_current_span's exception handling and not formatting the description properly. After removing the duplicate code, had to update the assertion to expect description formatted by the context manager. Similar changes below.

**Review by codeboten — APPROVED** [review]
nice, this cleans up the code quite a bit

**Pull request merged**